libs/unity/library/Tests/Runtime/PeerConnectionTests.cs

```diff
@@ -28,25 +73,118 @@ public IEnumerator SimplePeerConnection()
-            yield return null;
+            // Create the signaler
+            var sig_go = new GameObject("signaler");
+            var sig = sig_go.AddComponent<LocalOnlySignaler>();
+            sig.Peer1 = pc1;
+            sig.Peer2 = pc2;
 
-            // Check the event was raised
-            Assert.IsTrue(initializedEvent.Wait(millisecondsTimeout: 50000));
-            Assert.IsNotNull(pc.Peer);
+            // Create the video source on peer #1
+            VideoTrackSource source1 = pc1_go.AddComponent<UniformColorVideoSource>();
+            VideoReceiver receiver1 = pc1_go.AddComponent<VideoReceiver>();
+            MediaLine ml1 = pc1.AddMediaLine(MediaKind.Video);
+            ml1.SenderTrackName = "video_track_1";
+            ml1.Source = source1;
+            ml1.Receiver = receiver1;
+
+            // Create the video source on peer #2
+            VideoTrackSource source2 = pc2_go.AddComponent<UniformColorVideoSource>();
+            VideoReceiver receiver2 = pc2_go.AddComponent<VideoReceiver>();
+            MediaLine ml2 = pc2.AddMediaLine(MediaKind.Video);
+            ml2.SenderTrackName = "video_track_2";
+            ml2.Source = source2;
+            ml2.Receiver = receiver2;
+
+            // Init/quit twice.
+            for (int i = 0; i < 2; ++i)
+            {
+                // Initialize
+                var init1 = InitializeAndWait(pc1);
+                var init2 = InitializeAndWait(pc2);
+                yield return init1;
+                yield return init2;
+
+                // Confirm the sources are ready.
+                Assert.IsTrue(source1.IsLive);
+                Assert.IsTrue(source2.IsLive);
+
+                // Sender tracks will be created on connection.
+                Assert.IsNull(ml1.SenderTrack);
+                Assert.IsNull(ml2.SenderTrack);
+
+                // Connect
+                Assert.IsTrue(sig.StartConnection());
+                yield return sig.WaitForConnection(millisecondsTimeout: 10000);
+                Assert.IsTrue(sig.IsConnected);
+
+                // Wait a frame so that the Unity events for streams started can propagate
+                yield return null;
+
+                // Check pairing
+                Assert.IsNotNull(receiver1.Transceiver);
+                Assert.IsTrue(receiver1.IsLive);
+                Assert.AreEqual(1, pc1.Peer.RemoteVideoTracks.Count());
+                Assert.IsNotNull(receiver2.Transceiver);
+                Assert.IsTrue(receiver2.IsLive);
+                Assert.AreEqual(1, pc2.Peer.RemoteVideoTracks.Count());
+
+                // Shutdown peer #1
+                pc1.enabled = false;
+                Assert.IsNull(pc1.Peer);
+
+                // The source is not impacted, but tracks and transceiver are gone.
+                Assert.IsTrue(source1.IsLive);
+                Assert.IsFalse(receiver1.IsLive);
+                Assert.IsNull(ml1.SenderTrack);
+                Assert.IsNull(ml1.Transceiver);
+
```

Comment:
Add some checks about the influence of shutting down the remote peer connection but not the local one? I assume the connection will be closed, but transceivers may still exist as well as tracks. Though I don't know if there's a good way to be notified when the connection is closed, and this might only be timeout-based, which is difficult to test.

Reply:
I tried waiting for a frame and the local peer (not shutdown) didn't report any change. So yeah, I am not sure if there is a way to reliably detect disconnection. I'll leave it like this and add a comment.